crypto/keyring/keyring.go

```diff
@@ -122,11 +125,13 @@ type Importer interface {
 	ImportPubKey(uid string, armor string) error
 }
 
-// LegacyInfoImporter is implemented by key stores that support import of Info types.
-type LegacyInfoImporter interface {
-	// ImportInfo import a keyring.Info into the current keyring.
-	// It is used to migrate multisig, ledger, and public key Info structure.
-	ImportInfo(oldInfo Info) error
+type Migrator interface {
+	MigrateAll() (bool, error)
+}
+
+type Setter interface {
+	// Set item in keyring.Item
+	SetItem(item keyring.Item) error
```

Comment:
Since `SetItem` is used  in tests, `keyring` and `keys` package, i decided to declare `Setter` interface and make `SetItem` public.